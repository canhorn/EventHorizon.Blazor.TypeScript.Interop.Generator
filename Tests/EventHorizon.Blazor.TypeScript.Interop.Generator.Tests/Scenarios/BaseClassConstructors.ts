declare module Examples {
  export class UsedClass {}
  export class FifthLayerBaseClass {
    constructor(name: string, usedClass?: Nullable<UsedClass>, isPure?: boolean);
  }
  export class FourthLayerBaseClass extends FifthLayerBaseClass {
    constructor(name: string, usedClass?: Nullable<UsedClass>, isPure: boolean);
  }
  export abstract class ThirdLayerBaseClass extends FourthLayerBaseClass {
    constructor(name: string, usedClass?: Nullable<UsedClass>);
  }
  export class SecondLayerBaseClass extends ThirdLayerBaseClass {
    constructor(
      name: string,
      usedClass?: Nullable<UsedClass>,
      parent?: Nullable<ThirdLayerBaseClass>,
      source?: Nullable<ExampleClass>,
      boolParam?: boolean,
      boolParam2?: boolean
    );
  }
  export class ExampleClass extends SecondLayerBaseClass {
    constructor(name: string, usedClass?: UsedClass);
  }
}
