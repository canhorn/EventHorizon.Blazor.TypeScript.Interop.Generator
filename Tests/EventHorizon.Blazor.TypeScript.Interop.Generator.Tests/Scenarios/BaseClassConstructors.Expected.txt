/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : SecondLayerBaseClass
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
        }

        public ExampleClass(
            string name, UsedClass usedClass = null
        ) : base()
        {
            var entity = EventHorizonBlazorInterop.New(
                new string[] { "Examples", "ExampleClass" },
                name, usedClass
            );
            ___guid = entity.___guid;
        }

        public ExampleClass(
            string name, UsedClass usedClass = null, ThirdLayerBaseClass parent = null, ExampleClass source = null, System.Nullable<bool> boolParam = null, System.Nullable<bool> boolParam2 = null
        ) : base()
        {
            var entity = EventHorizonBlazorInterop.New(
                new string[] { "Examples", "ExampleClass" },
                name, usedClass, parent, source, boolParam, boolParam2
            );
            ___guid = entity.___guid;
        }

        public ExampleClass(
            string name, bool isPure, UsedClass usedClass = null
        ) : base()
        {
            var entity = EventHorizonBlazorInterop.New(
                new string[] { "Examples", "ExampleClass" },
                name, usedClass, isPure
            );
            ___guid = entity.___guid;
        }
        #endregion

        #region Methods

        #endregion
    }
}